### ISSUE TITLE
Error in IDE - java.nio.file.FileSystemException: /fragment.kt: Read-only file system

### ISSUE THREAD

**Issue by pinkasey** [detekt/detekt-intellij-plugin#80]
It happened to me this morning, 39 times apparently.
I noticed only 34 minutes later.
I don't know of something special that I did at the time - I didn't checkout branches or something like that.
Looks like the plugin has no access to write a tmp file.

BTW,
I have the plugin configured with my own private jars ("Plugin jars" configuration), but I doubt it was the source of problem this time.

Here's the stack-trace:
```
ExternalToolPass: 

com.intellij.diagnostic.PluginException: annotator: io.gitlab.arturbosch.detekt.DetektAnnotator@fbdbe18 (class io.gitlab.arturbosch.detekt.DetektAnnotator) [Plugin: detekt]
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:405)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:58)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.process(ExternalToolPass.java:253)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:206)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.access$500(ExternalToolPass.java:39)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.lambda$run$1(ExternalToolPass.java:179)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.lambda$runUnderDisposeAwareIndicator$11(BackgroundTaskUtil.java:243)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$3(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:163)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:585)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:150)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:263)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:242)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass$1.run(ExternalToolPass.java:178)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:324)
	at com.intellij.util.ui.update.MergingUpdateQueue.execute(MergingUpdateQueue.java:314)
	at com.intellij.util.ui.update.MergingUpdateQueue.lambda$flush$1(MergingUpdateQueue.java:264)
	at com.intellij.util.ui.update.MergingUpdateQueue.flush(MergingUpdateQueue.java:278)
	at com.intellij.util.ui.update.MergingUpdateQueue.run(MergingUpdateQueue.java:233)
	at com.intellij.util.concurrency.QueueProcessor.runSafely(QueueProcessor.java:231)
	at com.intellij.util.Alarm$Request.runSafely(Alarm.java:371)
	at com.intellij.util.Alarm$Request.run(Alarm.java:358)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.SchedulingWrapper$MyScheduledFutureTask.run(SchedulingWrapper.java:217)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:222)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:30)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.execute(BoundedTaskExecutor.java:201)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:221)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:190)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.nio.file.FileSystemException: /fragment.kt: Read-only file system
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:100)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:219)
	at java.base/java.nio.file.spi.FileSystemProvider.newOutputStream(FileSystemProvider.java:478)
	at java.base/java.nio.file.Files.newOutputStream(Files.java:219)
	at com.intellij.util.io.PathKt.outputStream(path.kt:48)
	at com.intellij.util.io.PathKt.write(path.kt:155)
	at com.intellij.util.io.PathKt.write$default(path.kt:154)
	at io.gitlab.arturbosch.detekt.util.PsiFileUtilsKt.toTmpFile(PsiFileUtils.kt:22)
	at io.gitlab.arturbosch.detekt.DetektAnnotator.runDetekt(DetektAnnotator.kt:40)
	at io.gitlab.arturbosch.detekt.DetektAnnotator.doAnnotate(DetektAnnotator.kt:31)
	at io.gitlab.arturbosch.detekt.DetektAnnotator.doAnnotate(DetektAnnotator.kt:22)
	at com.intellij.codeInsight.daemon.impl.ExternalToolPass.doAnnotate(ExternalToolPass.java:203)
	... 32 more
```

**Comment by arturbosch** [comment]
Hey,
Do you have a special setup or something where writing to tmp is not allowed?
Do you remember which idea features you used to get these ?
I assume you use the newest version available?

**Comment by pinkasey** [comment]
No, I don't think I have anything special about my setup or tmp directory.
I've installed detekt-plugin a few weeks ago, works great so far.
Really, this crash was the first time I saw anything wrong, and even then - I wouldn't have noticed it unless there was a balloon.

I'll post the version I have installed tomorrow.

**Comment by pinkasey** [comment]
I don't think it's worth digging into, I just posted it to have record of it.
If it happens again, I'll post a comment.

**Comment by arturbosch** [comment]
Thanks for the insights.
Have you updated to 1.0.0 meanwhile? 
This crash with `/fragment.kt` could have been fixed in 1.0.0 as I faced a similar issue when writing tests for this plugin.

**Comment by pinkasey** [comment]
I updated yesterday, following your comment,
I'll let you know if it happens again.
